agent/utils/utils.go

```diff
@@ -46,6 +47,10 @@ func ZeroOrNil(obj interface{}) bool {
 	if obj == nil {
 		return true
 	}
+	str, _ := json.Marshal(obj)
+	if string(str) == "null" {
```

Comment:
when would this condition get hit? afaik "null" should never show up here.

Reply:
nvm -- i see it

Reply:
Also -- you shouldn't be attempting to json.Marshal anything that comes through here. That likely isn't safe. `ZeroOrNil` is probably the wrong function to depend on for what you're doing.

Reply:
it's comparing the marshalled value not the string "null" value itself. I think I'm getting myself confused now, am looking into it.

Reply:
So in our unmarshaler, we try to handle all values that mean "not set" from json's perspective: null, 3, or also not present.

Once we marshal it in though, it has the value `3` meaning not set.

Another way to think about this -- the struct we're working with doesn't have a zero value, so calling "ZeroOrNil" isn't designed to work here:

```
        _ Conditional = iota  // this would have been the zero value, but we don't have it.
	ExplicitlyEnabled
	ExplicitlyDisabled
	NotSet
```

Reply:
Hence the update in zeroOrNil function. 
Marshal/unmarshal logic didn't change from when it was just Conditional for TaskCPUMemLimit, which would suggest that merge logic doesn't work for this config as-is either. I'm taking a look.

Reply:
> Hence the update in zeroOrNil function.

You shouldn't use or update this function, you are updating it to hit a corner case for something that doesn't apply.

Reply:
This function is already used by the merge logic though. Again, I'm looking into exactly how TaskCPUMemLimit was working as expected with merge.

Reply:
I understand that merge currently uses that. But you should be performing this evaluation in your merge code instead of adding subtle check in the utility function that works only for the specific use case you're currently dealing with.

The right way to do that is definitely not by changing the definition of what ZeroOrNil means.